sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -435,6 +435,11 @@ object SQLConf {
     defaultValue = Some(true),
     doc = "When false, we will treat bucketed table as normal table")
 
+  val ORDER_BY_ORDINAL = booleanConf("spark.sql.orderByOrdinal",
```

Comment:
There's a similar configuration value in hive since 0.11 and later, which is ` hive.groupby.orderby.position.alias`(default as `false`).
See: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+SortBy#LanguageManualSortBy-SyntaxofOrderBy

I think we also need to read from ConfVar for this config.